### PR TITLE
Database

### PR DESCRIPTION
**Pull request by Bulldog2740**

### CODE REVIEW

File: src/main/java/com/util/Close.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
+package com.mateacademy;
+
+import java.sql.Connection;
+import java.sql.Statement;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.sql.ResultSet;
+
+public class Close {
```
Comment (Taniann) [inline src/main/java/com/util/Close.java]: [Class names are typically nouns or noun phrases](https://google.github.io/styleguide/javaguide.html#s5.2.2-class-names)

File: src/main/java/com/util/Close.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    public static void closer(Connection connection, PreparedStatement preparedStatement) throws SQLException {
```
Comment (Taniann) [inline src/main/java/com/util/Close.java]: [Method names are typically verbs or verb phrases.](https://google.github.io/styleguide/javaguide.html#s5.2.3-method-names)

File: src/main/java/com/util/Connector.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+import java.sql.SQLException;
+
+public class Connector {
+    private static String url = "jdbc:mysql://localhost:3306/homework" +
+            "?useUnicode=true&useJDBCCompliantTimezoneShift=true&useLegacyDatetimeCode=false&serverTimezone=UTC";
+    private static String user = "root";
+    private static String password = "root";
+
+    @SneakyThrows(SQLException.class)
+    public static Connection getConnnect() {
```
Comment (Taniann) [inline src/main/java/com/util/Connector.java]: better name getConnection()

File: src/main/java/com/util/Connector.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
+package com.mateacademy;
+
+import lombok.SneakyThrows;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+
+public class Connector {
```
Comment (Taniann) [inline src/main/java/com/util/Connector.java]: better make singleton class

File: src/main/resources/sql/homework_init.sql [hunk 5]
```diff
@@ -0,0 +1,131 @@
+CREATE SCHEMA `homework` ;
+USE homework;
+
+CREATE TABLE `homework`.`developers` (`id` INT NOT NULL AUTO_INCREMENT,
+
+`name` VARCHAR(255) NOT NULL,
+
```
Comment (Taniann) [inline src/main/resources/sql/homework_init.sql]: remove redundant blank line

Comment (Taniann) [reply src/main/resources/sql/homework_init.sql]: in whole file

File: src/test/java/mateacademy/CrudTest.java [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+import org.junit.Test;
+
+import java.sql.SQLException;
+
+public class CrudTest {
+    private static Logger logger = Logger.getLogger(CrudTest.class);
+    private CrudTask crud = new CrudTask();
+
+    @Test
+    public void shouldCreateDataInDataBaseInChosenTable() {
```
Comment (Taniann) [inline src/test/java/mateacademy/CrudTest.java]: you actually dont assert anything in these methods

File: src/main/java/com/CrudTask.java [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+import org.apache.log4j.Logger;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
+import java.sql.SQLException;
+
+public class CrudTask {
+    private static Logger logger = Logger.getLogger(CrudTask.class);
```
Comment (Taniann) [inline src/main/java/com/CrudTask.java]: better also make it final
and name for constants in UPPER_CASE

File: src/main/java/com/CrudTask.java [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+
+import org.apache.log4j.Logger;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
+import java.sql.SQLException;
+
+public class CrudTask {
```
Comment (Taniann) [inline src/main/java/com/CrudTask.java]: create separate repository class for each table

File: src/main/java/com/CrudTask.java [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            logger.error("SQLException: " + e);
+        }
+    }
+
+    public void readData(String tableName) throws SQLException {
+        Connection connection = Connector.getConnnection();
+        ResultSet resultSet;
+        String query = "SELECT * FROM" + tableName;
```
Comment (Taniann) [inline src/main/java/com/CrudTask.java]: use constants for queries

File: src/main/java/com/CrudTask.java [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+        } else throw new IllegalArgumentException();
+        try (Connection connection = Connector.getConnnection();
+             PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            logger.error("SQLException: " + e);
+        }
+    }
+
+    public void readData(String tableName) throws SQLException {
```
Comment (Taniann) [inline src/main/java/com/CrudTask.java]: why readData returns nothing

File: src/main/java/com/util/Connector.java (hunk 4 above)
Comment (Taniann) [inline src/main/java/com/util/Connector.java]: it is util class so better make it with private constructor and make it final

File: src/main/java/com/util/Connector.java (hunk 3 above)
Comment (Bulldog2740) [reply src/main/java/com/util/Connector.java]: okey

File: src/main/java/com/crud/CustomersTask.java [hunk 11]
```diff
@@ -0,0 +1,76 @@
...
+        } catch (SQLException e) {
+            LOGGER.error(e);
+        }
+    }
+
+    public void updateCustomer(Customer customer, Connection connection, int id) {
+
+        try (PreparedStatement statement = connection.prepareStatement(UPDATE)) {
+
+            statement.setInt(1, id);
```
Comment (Taniann) [inline src/main/java/com/crud/CustomersTask.java]: why do you also pass id
why dont you use customer.getId() ?

File: src/test/java/mateacademy/CrudTest.java [hunk 12]
```diff
@@ -0,0 +1,46 @@
+package java.mateacademy;
+
+import com.mateacademy.CrudTask;
+import org.apache.log4j.Logger;
+import org.junit.Test;
+
+import java.sql.SQLException;
+
+public class CrudTest {
```
Comment (Taniann) [inline src/test/java/mateacademy/CrudTest.java]: these tests are useless. the dont assert anything

File: src/main/java/com/crud/CompaniesTask.java [hunk 13]
```diff
@@ -0,0 +1,73 @@
...
+import java.util.List;
+
+public class CompaniesTask {
+    private static final Logger LOGGER = Logger.getLogger(CompaniesTask.class);
+    private static final String CREATE = "INSERT INTO companies (id, name, site,) VALUES (null, ?, ?)";
+    private static final String READ = "SELECT * FROM companies";
+    private static final String UPDATE = "UPDATE companies SET name = ?, site = ? WHERE id = ?";
+    private static final String DELETE = "DELETE FROM companies WHERE name = ?";
+    
+    public void addCompany(Company company, Connection connection) {
```
Comment (Taniann) [inline src/main/java/com/crud/CompaniesTask.java]: dont forget to close connection

File: src/test/java/mateacademy/CrudTest.java (hunk 12 above)
Comment (Bulldog2740) [reply src/test/java/mateacademy/CrudTest.java]: i delete them

File: src/main/java/com/crud/CompaniesTask.java [hunk 14]
```diff
@@ -0,0 +1,96 @@
...
+        try (PreparedStatement statement = connection.prepareStatement(CREATE)) {
+
+            statement.setString(2, company.getName());
+            statement.setString(3, company.getSite());
+            statement.executeUpdate();
+
+        } catch (SQLException e) {
+            LOGGER.error(e);
+        } finally {
+            statement.close();
```
Comment (Taniann) [inline src/main/java/com/crud/CompaniesTask.java]: you need to close connection
for statement you use try-with-resources

**Pull request merged**